es6-shim.js

```diff
@@ -1416,41 +1410,55 @@
       return ((al * bl) + (((ah * bl + al * bh) << 16) >>> 0) | 0);
     },
 
-    fround: function fround(x) {
-      var v = Number(x);
-      if (v === 0 || v === Infinity || v === -Infinity || numberIsNaN(v)) {
-        return v;
+    fround: function fround(value) {
+      var x = Number(value);
+      if (x === 0 || x === Infinity || x === -Infinity || Number.isNaN(x)) {
+        return x;
       }
-      var sign = Math.sign(v);
-      var abs = Math.abs(v);
+      var sign = Math.sign(x);
+      var abs = Math.abs(x);
       if (abs < BINARY_32_MIN_VALUE) {
         return sign * roundTiesToEven(abs / BINARY_32_MIN_VALUE / BINARY_32_EPSILON) * BINARY_32_MIN_VALUE * BINARY_32_EPSILON;
       }
       // Veltkamp's splitting (?)
       var a = (1 + BINARY_32_EPSILON / Number.EPSILON) * abs;
       var result = a - (a - abs);
-      if (result > BINARY_32_MAX_VALUE || numberIsNaN(result)) {
+      if (result > BINARY_32_MAX_VALUE || Number.isNaN(result)) {
         return sign * Infinity;
       }
       return sign * result;
     }
   };
   defineProperties(Math, MathShims);
-  // IE 11 TP has an imprecise log1p: reports Math.log1p(-1e-17) as 0
-  defineProperty(Math, 'log1p', MathShims.log1p, Math.log1p(-1e-17) !== -1e-17);
+
+  // https://code.google.com/p/v8/issues/detail?id=3468
+  var MATH_EXP_ERROR = Math.abs(1 - Math.exp(704.6589) / 1.0702171200481775e+306) / Math.pow(2, -52);
```

Comment:
Separate variables with a comment explaining exactly those things would be wonderful!